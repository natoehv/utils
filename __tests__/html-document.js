'use strict';

const document = require('../lib/html-document');

/**
 * .document()
 */

test('.document() - no arguments given - should render template', () => {
    const result = document();
    expect(result).toMatchSnapshot();
});

test('.document() - arguments given - should render template using values given', () => {
    const result = document({
        head: 'this goes in the head section',
        body: 'this goes in the body section',
        encoding: 'utf-pretend-encoding',
        locale: 'en-NZ',
        title: 'this goes in the title tag',
        js: 'http://somejsurl.com',
        css: 'http://somecssurl.com',
    });
    expect(result).toMatchSnapshot();
});

test('.document() - arguments given - handles v4 js and css syntax', () => {
    const result = document({
        js: [
            { value: 'http://somejsurl1.com', type: 'default' },
            { value: 'http://somejsurl2.com', type: 'default' },
            { value: 'http://somejsurl3.com', type: 'default' },
        ],
        css: [
            { value: 'http://somecssurl1.com', type: 'default' },
            { value: 'http://somecssurl2.com', type: 'default' },
            { value: 'http://somecssurl3.com', type: 'default' },
        ],
    });
    expect(result).toMatchSnapshot();
});

test('.document() - js "type" is "module" - should set type to module on script tags', () => {
    const result = document({
        js: [
            { value: 'http://somejsurl1.com', type: 'module' },
            { value: 'http://somejsurl2.com', type: 'module' },
            { value: 'http://somejsurl3.com', type: 'module' },
        ],
        css: [
            { value: 'http://somecssurl1.com', type: 'default' },
            { value: 'http://somecssurl2.com', type: 'default' },
            { value: 'http://somecssurl3.com', type: 'default' },
        ],
    });
    expect(result).toMatchSnapshot();
});