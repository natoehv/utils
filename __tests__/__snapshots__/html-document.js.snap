// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.document() - arguments given - handles v4 js and css syntax 1`] = `
"<!doctype html>
<html lang=\\"en-US\\">
    <head>
        <meta charset=\\"utf-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl1.com\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl2.com\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl3.com\\">
        <script defer src=\\"http://somejsurl1.com\\" ></script>
        <script defer src=\\"http://somejsurl2.com\\" ></script>
        <script defer src=\\"http://somejsurl3.com\\" ></script>
        <title></title>
        
    </head>
    <body>
        
    </body>
</html>"
`;

exports[`.document() - arguments given - should render template using values given 1`] = `
"<!doctype html>
<html lang=\\"en-NZ\\">
    <head>
        <meta charset=\\"utf-pretend-encoding\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl.com\\">
        <script defer src=\\"http://somejsurl.com\\" ></script>
        <title>this goes in the title tag</title>
        this goes in the head section
    </head>
    <body>
        this goes in the body section
    </body>
</html>"
`;

exports[`.document() - js "type" is "esm" - should set type to module on script tags 1`] = `
"<!doctype html>
<html lang=\\"en-US\\">
    <head>
        <meta charset=\\"utf-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl1.com\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl2.com\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl3.com\\">
        <script type=\\"module\\" defer src=\\"http://somejsurl1.com\\" ></script>
        <script type=\\"module\\" defer src=\\"http://somejsurl2.com\\" ></script>
        <script type=\\"module\\" defer src=\\"http://somejsurl3.com\\" ></script>
        <title></title>
        
    </head>
    <body>
        
    </body>
</html>"
`;

exports[`.document() - no arguments given - should render template 1`] = `
"<!doctype html>
<html lang=\\"en-US\\">
    <head>
        <meta charset=\\"utf-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        
        
        <title></title>
        
    </head>
    <body>
        
    </body>
</html>"
`;
